### PR TITLE
48: Add HeadingIcon component

### LINKED ISSUE #48: Add HeadingIcon component

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
# Done
- Added [HeadingIcon](https://docs.vanillaframework.io/en/patterns/heading-icon) component
- Added story to Storybook
- Added snapshot tests

# QA
- Pull code
- Run `yarn clean` and `yarn serve` and navigate to http://localhost:6006/
- Run `yarn lint` and `yarn test` to ensure there are no linting or testing errors
- Verify the HeadingIcon component in Storybook

# Fixes
Fixes #48

### CODE REVIEW

File: src/lib/components/HeadingIcon/HeadingIcon.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+@import 'vanilla-framework/scss/base';
+@import 'vanilla-framework/scss/patterns_heading-icon';
+@include vf-base;
+@include vf-p-heading-icon
```
Comment (tbille) [inline src/lib/components/HeadingIcon/HeadingIcon.scss]: `;` missing

File: src/lib/components/HeadingIcon/HeadingIcon.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './HeadingIcon.css';
+
+const HeadingIcon = props => (
+  <div className="p-heading-icon">
+    <div className="p-heading-icon__header">
+      <img className="p-heading-icon__img" alt="" src={props.imageSrc} />
```
Comment (barrymcgee) [inline src/lib/components/HeadingIcon/HeadingIcon.js]: `props.title` should be passed as an alt value or an additional prop added for the alt value.

**Review by barrymcgee — CHANGES_REQUESTED** [review]

**Comment by Caleb-Ellis** [comment]
@barrymcgee added optional alt prop with default empty string value

File: src/lib/components/HeadingIcon/HeadingIcon.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './HeadingIcon.css';
+
+const HeadingIcon = props => (
+  <div className="p-heading-icon">
+    <div className="p-heading-icon__header">
+      <img className="p-heading-icon__img" alt={props.alt} src={props.src} />
+      <h3 className="p-heading-icon__title">{ props.title }</h3>
```
Comment (anthonydillon) [inline src/lib/components/HeadingIcon/HeadingIcon.js]: Should this have space around the `{`?

**Review by anthonydillon — APPROVED** [review]
Small comment in code.

File: src/lib/components/HeadingIcon/HeadingIcon.js (hunk 3 above)
Comment (Caleb-Ellis) [reply src/lib/components/HeadingIcon/HeadingIcon.js]: @anthonydillon Hmm I guess not. I honestly thought it was a linting rule to have spaces on children, and no spaces everywhere else, but it doesn't catch it whether the space is there or not. I'll remove the spaces.

**Pull request merged**